src/rdb.c

```diff
@@ -2939,6 +2940,8 @@ void startSaving(int rdbflags) {
         subevent = REDISMODULE_SUBEVENT_PERSISTENCE_AOF_START;
     else if (rdbflags & RDBFLAGS_AOF_PREAMBLE)
         subevent = REDISMODULE_SUBEVENT_PERSISTENCE_SYNC_AOF_START;
+    else if (rdbflags & RDBFLAGS_REPLICATION)
+        subevent = REDISMODULE_SUBEVENT_PERSISTENCE_REPL_START;
```

Comment:
this could be a breaking change for some module.
i.e. one that waits for RDB_START for some reason, and concludes that's a full sync in another way.
@MeirShpilraien WDYT?

Reply:
Yes I agree it is a breaking change. Maybe we can use the extra data to indicate that the rdb is creating for replication?

Reply:
Sorry. I don't notice this. I can revert it if needed.

Reply:
maybe it's ok to break if we document it properly. let's consider our alternatives.

Reply:
ok, let's drop that part. and merge the rest as cleanup (without any interface changes).
if some module will require to know the purpose for the rdb saving in the future, we can then discuss the requirement and see how we solve it (possibly by passing some new struct to the `data` pointer).

p.s. when disk-based replication produces an rdb for replication, it'll also serve for persistence (same goes for diskless replica receiving it), 
so arguably, any module that uses that info may be doing something wrong.

Reply:
> if some module will require to know the purpose for the rdb saving in the future, we can then discuss the requirement and see how we solve it (possibly by passing some new struct to the `data` pointer).

Yes, I think this is a good idea. Users can use a callback function to distinguish the purpose.

> p.s. when disk-based replication produces an rdb for replication, it'll also serve for persistence (same goes for diskless replica receiving it),
> so arguably, any module that uses that info may be doing something wrong.

Yes, I agree with you. Unless the module doesn't care what triggers the rdb saving and just wants to know there is a rdb saving in progress.